lib/native/http.ts

```diff
@@ -23,74 +23,78 @@ export { HttpHeaders, HttpRequest } from '../common/http';
 import { BufferedEventEmitter } from '../common/event';
 
 /** Base class for HTTP connections */
-export class HttpConnection extends NativeResource implements ResourceSafe {
+export class HttpConnection extends NativeResourceMixin(BufferedEventEmitter) implements ResourceSafe {
 
     protected constructor(native_handle: any) {
-        super(native_handle);
+        super();
+        this._super(native_handle);
     }
 
     close() {
         crt_native.http_connection_close(this.native_handle());
     }
-}
 
-type ClientConnectionCallback = (connection: HttpClientConnection, error_code: Number) => void;
+    /** Emitted when the connection is connected and ready to start streams */
+    on(event: 'ready', listener: () => void): this;
```

Comment:
Not that we have to match Node exactly, but their HTTP/2 API calls this ['connect'](https://nodejs.org/api/http2.html#http2_event_connect)